h2o-algos/src/main/java/hex/mli/loco/LeaveOneCovarOut.java

```diff
@@ -0,0 +1,245 @@
+            tryComplete();
+        }
+
+    }
+
+    /**
+     * Get base predictions given a model and frame. "Base predictions" are predictions based on all features in the
+     * model
+     * @param m An H2O supervised model
+     * @param fr A Frame to score on
+     * @return An array of Vecs containing predictions
+     */
+    private static Vec[] getBasepredictions(Model m, Frame fr){
+
+        Frame basePredsFr = m.score(fr,null,null,false);
+
+        if(m._output.getModelCategory() == ModelCategory.Binomial) {
+            Vec basePreds = basePredsFr.remove(2);
+            basePredsFr.delete();
+            return new Vec[] {basePreds};
+        }else if(m._output.getModelCategory() == ModelCategory.Multinomial){
+            Vec[] basePredsVecs = basePredsFr.vecs();
+            DKV.remove(basePredsFr._key);
+            return basePredsVecs;
+        } else {
+            Vec basePreds = basePredsFr.remove(0);
+            basePredsFr.delete();
+            return new Vec[] {basePreds};
+        }
+
+    }
+
+    /**
+     * Get new predictions based on dropping/replacing a column with mean, median, or mode
+     * @param m An H2O supervised model
+     * @param fr A Frame to score on
+     * @param colToDrop Column to modify/drop before prediction
+     * @param valToReplace Value to replace colToDrop by (Default is null)
+     * @return
+     */
+    private static Vec[] getNewPredictions(Model m, Frame fr, String colToDrop, String valToReplace) {
+        Frame workerFrame = new Frame(fr);
+        Vec vecToReplace = fr.vec(colToDrop);
+        Vec replacementVec = null;
+        if(valToReplace == null){
+            replacementVec = vecToReplace.makeCon(Double.NaN);
+        } else if(valToReplace.equals("mean")){
+            if(vecToReplace.isCategorical()){
+                Vec tmpVec = vecToReplace.makeCon(vecToReplace.mode());
+                replacementVec = tmpVec.toCategoricalVec(); //Can only get mode for categoricals
+                tmpVec.remove();
+            } else {
+                replacementVec = vecToReplace.makeCon(vecToReplace.mean());
+            }
+        } else if(valToReplace.equals("median")){
+            if(vecToReplace.isCategorical()){
+                Vec tmpVec = vecToReplace.makeCon(vecToReplace.mode());
+                replacementVec = tmpVec.toCategoricalVec();  //Can only get mode for categoricals
+                tmpVec.remove();
+            } else {
+                Frame tmpFr = new Frame(vecToReplace);
+                double median = AstMedian.median(tmpFr, CombineMethod.AVERAGE);
+                replacementVec = vecToReplace.makeCon(median);
+            }
+        } else {
+            throw new H2OIllegalArgumentException("Invalid value to replace columns in LOCO. Got " + valToReplace);
+        }
+        int vecToDropIdx = fr.find(colToDrop);
+        workerFrame.replace(vecToDropIdx,replacementVec);
+        DKV.put(workerFrame);
```

Comment:
if you have job here, you can lock frame at creation, and then call `update` instead of `DKV.put`